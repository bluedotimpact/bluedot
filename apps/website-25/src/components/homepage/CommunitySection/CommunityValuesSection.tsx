import { Card, SlideList } from '@bluedot/ui';

const values = [
  {
    imageSrc: '/images/community-values/network.png',
    title: 'Our online community brings together 4,500+ professionals across 100+ countries',
  },
  {
    imageSrc: '/images/community-values/people.png',
    title: 'Diverse expertise, from engineers and entrepreneurs to policymakers and philosophers',
  },
  {
    imageSrc: '/images/community-values/study.png',
    title: '40+ independent study groups worldwide have used our materials to run their own courses',
  },
  {
    imageSrc: '/images/community-values/world.png',
    title: 'We run regular in-person events across the world, turning online connections into lasting relationships',
  },
] as const;

const CommunityValuesSection = () => {
  return (
    <SlideList title="Our community" maxItemsPerSlide={4} className="community-values-section pb-spacing-y">
      {values.map((value) => (
        <Card {...value} className="community-values-section__value" />
      ))}
    </SlideList>
  );
};

export default CommunityValuesSection;
