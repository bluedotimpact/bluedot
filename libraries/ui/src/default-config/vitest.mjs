/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
