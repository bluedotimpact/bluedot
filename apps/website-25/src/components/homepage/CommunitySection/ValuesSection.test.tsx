import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import CommunityValuesSection from './CommunityValuesSection';

describe('CommunityValuesSection', () => {
  test('renders default as expected', () => {
    const { container } = render(<CommunityValuesSection />);
    expect(container).toMatchSnapshot();
  });
});
