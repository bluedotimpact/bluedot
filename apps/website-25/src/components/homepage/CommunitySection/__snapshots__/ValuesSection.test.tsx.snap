// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommunityValuesSection > renders default as expected 1`] = `
<div>
  <div
    class="section w-full max-w-max-width min-w-min-width mx-auto overflow-y-scroll md:px-gutter px-gutter-sm my-8 community-values-section"
  >
    <div
      class="section__title-container flex justify-between items-center gap-4"
    >
      <div
        class="section__content flex-1"
      />
    </div>
    <div
      class="section__body"
    >
      <div
        class="community-values-section__container flex sm:flex-row flex-col gap-8 overflow-y-scroll"
      >
        <div
          class="value-card w-full flex flex-col gap-6 py-6"
        >
          <div
            class="value-card__icon-wrapper size-[50px] flex items-center justify-center"
          >
            <img
              class="value-card__icon"
              src="icons/network.svg"
            />
          </div>
          <p
            class="value-card__description font-semibold"
          >
            Our online community brings together 4,500+ professionals across 100+ countries
          </p>
        </div>
        <div
          class="value-card w-full flex flex-col gap-6 py-6"
        >
          <div
            class="value-card__icon-wrapper size-[50px] flex items-center justify-center"
          >
            <img
              class="value-card__icon"
              src="icons/people.svg"
            />
          </div>
          <p
            class="value-card__description font-semibold"
          >
            Diverse expertise, from engineers and entrepreneurs to policymakers and philosophers
          </p>
        </div>
        <div
          class="value-card w-full flex flex-col gap-6 py-6"
        >
          <div
            class="value-card__icon-wrapper size-[50px] flex items-center justify-center"
          >
            <img
              class="value-card__icon"
              src="icons/study.svg"
            />
          </div>
          <p
            class="value-card__description font-semibold"
          >
            40+ independent study groups worldwide have used our materials to run their own courses
          </p>
        </div>
        <div
          class="value-card w-full flex flex-col gap-6 py-6"
        >
          <div
            class="value-card__icon-wrapper size-[50px] flex items-center justify-center"
          >
            <img
              class="value-card__icon"
              src="icons/world.svg"
            />
          </div>
          <p
            class="value-card__description font-semibold"
          >
            We run regular in-person events across the world, turning online connections into lasting relationships
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
